src/Handler/CurlFactory.php

```diff
@@ -419,6 +423,27 @@ private function applyHandlerOptions(EasyHandle $easy, array &$conf)
             $conf[CURLOPT_VERBOSE] = true;
         }
     }
+    
+    private function isHostInProxyArea($host, array $areas)
+    {
+        $matches = false;
+        foreach ($areas as $area) {
+            $areaToMatch = $area;
+            // if include all subdomains 
+            if (!empty($area) && substr($area, 0, 1) === '.'){
+                $areaToMatch = "*$area";
+            }
+            // Use wilcards
+            $delimiter = '#';
```

Comment:
The need is to escape regex characters that we can have in the hostname and take in consideration the wilcard.